sdk/clientcore/core/src/main/java/io/clientcore/core/http/pipeline/HttpPipelineOrder.java

```diff
@@ -0,0 +1,160 @@
+     *     +-------------------+
+     * </pre>
+     */
+    public static final HttpPipelineOrder BETWEEN_REDIRECT_AND_RETRY
+        = new HttpPipelineOrder("BETWEEN_REDIRECT_AND_RETRY");
+
+    /**
+     * The policy will be position after the {@link HttpRetryPolicy} and before the {@link HttpCredentialPolicy}.
+     * <p>
+     * If multiple {@link HttpPipelinePolicy policies} are added with this order, they will be executed in the order
+     * they were added between the {@link HttpRetryPolicy} and {@link HttpCredentialPolicy}. Or, visually:
+     * <pre>
+     *     +-----------------------+
+     *     |     Retry Policy      |
+     *     +-----------------------+
+     *     |   1st added Between   |
+     *     +-----------------------+
+     *     |   2nd added Between   |
+     *     +-----------------------+
+     *     |   3rd added Between   |
+     *     +-----------------------+
+     *     | Authentication Policy |
+     *     +-----------------------+
+     * </pre>
+     */
+    public static final HttpPipelineOrder BETWEEN_RETRY_AND_AUTHENTICATION
+        = new HttpPipelineOrder("BETWEEN_RETRY_AND_AUTHENTICATION");
+
+    /**
+     * The policy will be position after the {@link HttpCredentialPolicy} and before the
+     * {@link HttpInstrumentationPolicy}.
+     * <p>
+     * If multiple {@link HttpPipelinePolicy policies} are added with this order, they will be executed in the order
+     * they were added between the {@link HttpCredentialPolicy} and {@link HttpInstrumentationPolicy}. Or, visually:
+     * <pre>
+     *     +------------------------+
+     *     | Authentication Policy  |
+     *     +------------------------+
+     *     |   1st added Between    |
+     *     +------------------------+
+     *     |   2nd added Between    |
+     *     +------------------------+
+     *     |   3rd added Between    |
+     *     +------------------------+
+     *     | Instrumentation Policy |
+     *     +------------------------+
+     * </pre>
+     */
+    public static final HttpPipelineOrder BETWEEN_AUTHENTICATION_AND_INSTRUMENTATION
+        = new HttpPipelineOrder("BETWEEN_AUTHENTICATION_AND_INSTRUMENTATION");
+
+    /**
+     * The policy will be position after the {@link HttpInstrumentationPolicy}.
+     * <p>
+     * If multiple {@link HttpPipelinePolicy policies} are added with this order, they will be executed in the order
+     * they were added after {@link HttpInstrumentationPolicy}. Or, visually:
+     * <pre>
+     *     +------------------------+
+     *     | Instrumentation Policy |
+     *     +------------------------+
+     *     |    1st added After     |
+     *     +------------------------+
+     *     |    2nd added After     |
+     *     +------------------------+
+     *     |    3rd added After     |
+     *     +------------------------+
+     * </pre>
+     */
+    public static final HttpPipelineOrder AFTER_INSTRUMENTATION = new HttpPipelineOrder("AFTER_INSTRUMENTATION");
```

Comment:
Policies added after instrumentation can potentially change the request resulting in a discrepancy between what is instrumented (logged) and what is actually sent over the wire. So, we should not support `AFTER_INSTRUMENTATION `. We can always add this later if we find a valid usecase for this.

Reply:
I added this even though we discussed not to as a test added by @lmolkova depended on this functionality. @lmolkova, would you consider that test a valid case for this? It seems the span, and other related telemetry / metrics stuff, doesn't exist until after this policy and is enriching it a case we see often?

Reply:
I'd like to be able to customize instrumentation policy and the next policy after could do the trick, but there are other ways to solve it.

The easiest one and the one I would prefer the most would be  to make HttpInstrPolicy not final and let users/SDKs extend it. I left another comment on it and other being final.

Reply:
BTW, circuit breaker or local cache could be an excellent candidate for the last policy.

Reply:
also fault injector, but that one should probably be done in the http client